trace/trace.go

```diff
@@ -206,6 +206,21 @@ func startSpanInternal(name string, hasParent bool, parent SpanContext, remotePa
 	span.spanContext = parent
 
 	cfg := config.Load().(*Config)
+	gen := cfg.IDGenerator.(*defaultIDGenerator)
```

Comment:
Please note that the `IDGenerator` can be overridden with a custom implementation, in which case this line would panic.

The easiest fix would be to perform a safe cast:
```go
gen, ok := cfg.IDGenerator.(*defaultIDGenerator)
if ok && !gen.randomIDsGenerated {
  ...
}
```

Reply:
See updated code example in the other comment.